src/test/assignthree/GraphTest.java

```diff
@@ -0,0 +1,84 @@
+package test.assignthree;
+
+import assignthree.model.Graph;
+import assignthree.model.Node;
+import org.junit.Test;
+
+import java.util.ArrayList;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+import static org.junit.Assert.*;
+
+/*
+ * It is use to test all test the function on data structure
+ */
+public class GraphTest {
+    @Test
+    /*
+     * unit testing
```

Comment:
what is unit testing ? what cases is it covering ?

Reply:
Zero negative cases

Reply:
ok